server/src/main/java/org/opensearch/common/xcontent/JsonToStringXContentParser.java

```diff
@@ -74,7 +85,15 @@ public XContentParser parseObject() throws IOException {
         return JsonXContent.jsonXContent.createParser(this.xContentRegistry, this.deprecationHandler, String.valueOf(jString));
     }
 
-    private void parseToken(StringBuilder path, String currentFieldName) throws IOException {
+    private void parseToken(StringBuilder path, String currentFieldName, int depth) throws IOException {
+        if (depth >= depthLimit) {
```

Comment:
@mingshl  I'm not sure whether it should throw exception or just ignore.

Reply:
the dept limit should be set to avoid the object being deeply nested. An exception is expected in this case. 

But ignoreAbove should not throw exceptions.